morse.rb

```diff
@@ -0,0 +1,51 @@
+@dictionary = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z'
+}
+
+def decode_char(letter)
+  @letter = @dictionary[letter]
+  @letter
+end
+
+def decode_word(word)
+  @word_array = word.split
+  @current_char = ''
+  @word_array.each do |one_word|
+    @current_char += decode_char(one_word)
+  end
+  print "#{@current_char} "
+end
+
+def decode_message(message)
+  @message_arr = message.split('   ')
+  @message_arr.each do |msg|
+    decode_word(msg)
+  end
+end
```

Comment:
- [ ] Your sentence decoder does not work as expected. After looping and decoding the message you still need to return the entire decoded message. Kindly do that. You can change to this:

  ```
  def decode_message(message)
    @message_arr = message.split('   ')
    decoded_msg = ''
    @message_arr.each do |msg|
      decoded_msg += decode_word(msg) + ' '
    end
    return decoded_msg.strip
  end
  ```

  ![ryru](https://user-images.githubusercontent.com/81468675/198003417-afb10ec3-53d3-421a-a787-3f8c0e3f8a7d.jpg)